packages/components/src/components/SquiggleViewer/SqViewNode.tsx

```diff
@@ -0,0 +1,166 @@
+import { SqValue, SqValuePath } from "@quri/squiggle-lang";
+
+import { getChildrenValues, TraverseCalculatorEdge } from "./utils.js";
+
+//We might want to bring this into the SquiggleLang library. The ``traverseCalculatorEdge`` part is awkward though.
+class SqValueNode {
+  constructor(
+    public root: SqValue,
+    public path: SqValuePath,
+    public traverseCalculatorEdge: TraverseCalculatorEdge
+  ) {}
+
+  uid() {
+    return this.path.uid();
+  }
+
+  isEqual = (other: SqValueNode): boolean => {
```

Comment:
Pseudo-methods like this won't always work correctly; tbh I don't remember the details but I remember I tried to use this pattern a while ago, but got burned and cached the belief that I should never do this.

Maybe it breaks the inheritance?.. I'm not sure.

Alternative solutions:
1) just inline it
2) never pass `obj.method` around as a function; pass `param => obj.method(param)` instead
3) explicitly `.bind` it in the constructor: `this.isEqual = this.isEqual.bind(this);`
4) Bind it at the place where the method is passed: `...findIndex(this.isEqual.bind(this));`

Out of these, (3) was a common practice back in class components days, but it's controversial because it's an action at distance and you have to remember which methods are rebound; (1), (2) and (4) are all fine by me.

Reply:
When you say ``inline it``, I take it you mean make it a normal method? Like,

```

  uid() {
    return this.path.uid();
  }

  isEqual(other: SqValueNode): boolean {
    return this.uid() === other.uid();
  }

  sqValue(): SqValue | undefined {
    return this.root.getSubvalueByPath(this.path, this.traverseCalculatorEdge);
  }
```

Reply:
That by itself is not enough.

You use `isEqual` only in `getParentIndex`, so (1) "inline it" is:

```typescript
  getParentIndex() {
    return this.siblings().findIndex(other => this.uid() === other.uid());
  }
```

(2) is:

```typescript
  isEqual(other: SqValueNode): boolean {
    return this.uid() === other.uid();
  }

  getParentIndex() {
    return this.siblings().findIndex(other => this.isEqual(other));
  }
```

And (4) is:
```typescript
  isEqual(other: SqValueNode): boolean {
    return this.uid() === other.uid();
  }

  getParentIndex() {
    return this.siblings().findIndex(this.isEqual.bind(this));
  }
```